### PR TITLE
Remove converter dir

### LINKED ISSUE #420: Remove converter directory
Since we're going to have a lot more eyes on this repo after glTF 1.0, we should tidy it up a bit.

Can we move `converter/COLLADA2GLTF` to `COLLADA2GLTF` and then remove the empty `converter` directory?

@tfili is this hard?  I think the only potential issue is submodules and updating any links in the README.md and on the wiki.

### PR DESCRIPTION
**Pull request by tfili**
Fixes #420. I made a pull request into dev-1.0 for this so someone else can make sure if you pull to an existing local repo it will still work.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Any links in README.md that need to be updated?

**Comment by tfili** [comment]
@pjcozzi. Not in the readme. There are some changes that need to be made in the Wiki, but that isn't branch specific so we'll have to wait till dev-1.0 goes in. The Wiki is so out of date anyway, it'll need a complete overhaul.

**Comment by tfili** [comment]
@pjcozzi Also don't forget a `git submodule sync` & `git submodule update`.

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "converter/COLLADA2GLTF/dependencies/OpenCOLLADA"]
 	path = converter/COLLADA2GLTF/dependencies/OpenCOLLADA
 	url = https://github.com/KhronosGroup/OpenCOLLADA.git
+[submodule "COLLADA2GLTF/dependencies/OpenCOLLADA"]
```
Comment (emackey) [inline .gitmodules]: Looks like the new location was added to `.gitmodules` without removing the old location?

Comment (tfili) [inline .gitmodules]: Hmmm. Weird. I thought the submodule remove we have done that. Should be fixed.

**Comment by pjcozzi** [comment]
Using a fresh clone, I get the following.  Looks like the submodule is not updated.  By the way, how valuable is it to have this as a submodule?  Why not just include all dependencies to avoid users needing to deal with submodules completely?

```
pcozzi@C1200906-00 /c/source
$ git clone --recurse-submodules https://github.com/KhronosGroup/glTF.git
Cloning into 'glTF'...
remote: Counting objects: 11308, done.
remote: Compressing objects: 100% (142/142), done.
Receiving objects: 100% (11308/11308), 39.70 MiB | 1.60 MiB/s, done.
emote: Total 11308 (delta 22), reused 0 (delta 0), pack-reused 11149
Resolving deltas: 100% (6474/6474), done.
Checking connectivity... done.
Submodule 'converter/COLLADA2GLTF/dependencies/OpenCOLLADA' (https://github.com/
KhronosGroup/OpenCOLLADA.git) registered for path 'converter/COLLADA2GLTF/depend
encies/OpenCOLLADA'
Submodule 'specification/validator/JSV' (https://github.com/garycourt/JSV.git) r
egistered for path 'specification/validator/JSV'
Cloning into 'converter/COLLADA2GLTF/dependencies/OpenCOLLADA'...
remote: Counting objects: 23968, done.
remote: Total 23968 (delta 0), reused 0 (delta 0), pack-reused 23968
Receiving objects: 100% (23968/23968), 20.14 MiB | 1.37 MiB/s, done.
Resolving deltas: 100% (18564/18564), done.
Checking connectivity... done.
Submodule path 'converter/COLLADA2GLTF/dependencies/OpenCOLLADA': checked out '4
924ae326b9e5e5874ae3ce37e0ff88cfa6bee68'
Cloning into 'specification/validator/JSV'...
remote: Counting objects: 597, done.
remote: Total 597 (delta 0), reused 0 (delta 0), pack-reused 597
Receiving objects: 100% (597/597), 434.32 KiB | 0 bytes/s, done.
Resolving deltas: 100% (304/304), done.
Checking connectivity... done.
Submodule path 'specification/validator/JSV': checked out 'e539d4d9e03785bd37159
ebf9f7169d9d3ae85ae'

pcozzi@C1200906-00 /c/source
$ cd glTF/

pcozzi@C1200906-00 /c/source/glTF (master)
$ git checkout remove-converter-dir
warning: unable to rmdir converter/COLLADA2GLTF/dependencies/OpenCOLLADA: Direct
ory not empty
Branch remove-converter-dir set up to track remote branch remove-converter-dir f
rom origin.
Switched to a new branch 'remove-converter-dir'

pcozzi@C1200906-00 /c/source/glTF (remove-converter-dir)
$ git submodule sync
Synchronizing submodule url for 'specification/validator/JSV'

pcozzi@C1200906-00 /c/source/glTF (remove-converter-dir)
$ git submodule update

pcozzi@C1200906-00 /c/source/glTF (remove-converter-dir)
$ cd COLLADA2GLTF/

pcozzi@C1200906-00 /c/source/glTF/COLLADA2GLTF (remove-converter-dir)
$ cmake CMakeLists.txt
-- Building for: Visual Studio 12 2013
-- The C compiler identification is MSVC 18.0.21005.1
-- The CXX compiler identification is MSVC 18.0.21005.1
-- Check for working C compiler using: Visual Studio 12 2013
-- Check for working C compiler using: Visual Studio 12 2013 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler using: Visual Studio 12 2013
-- Check for working CXX compiler using: Visual Studio 12 2013 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find ZLIB (missing:  ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
-- Could NOT find PNG (missing:  PNG_LIBRARY PNG_PNG_INCLUDE_DIR)
CMake Warning at CMakeLists.txt:76 (message):
  libpng or one of its dependencies couldn't be found.  Transparency may not
  be correctly detected.


CMake Error at CMakeLists.txt:87 (add_subdirectory):
  The source directory

    C:/source/glTF/COLLADA2GLTF/dependencies/OpenCOLLADA

  does not contain a CMakeLists.txt file.


[o3dgc]          LIB_TYPE STATIC
-- Configuring incomplete, errors occurred!
See also "C:/source/glTF/COLLADA2GLTF/CMakeFiles/CMakeOutput.log".
```

**Comment by tfili** [comment]
The following commands are what I had to run to get it working with a fresh clone. I forgot the `--init`, since you didn't have a OpenCollada directory in the new spot. The only issue we have really had with submodules is when we move them, and that is only temporary.
* `git clone --recurse-submodules git@github.com:KhronosGroup/glTF.git`
* `cd glTF`
* `git checkout remove-converter-dir`
* `rm -rf converter/`
* `git submodule sync`
* `git submodule update --init`
* `cd COLLADA2GLTF/`
* `cmake -G "Visual Studio 12 Win64"`

Once this change is merged in, none of this will be necessary. The `git clone` will be all that is needed for fresh repos and the sync & update will be needed with existing repos (just like any other submodule change).

**Comment by pjcozzi** [comment]
Thanks @tfili

**Pull request merged**